src/pymor/algorithms/lradi.py

```diff
@@ -194,14 +190,15 @@ def projection_shifts_init(A, E, B, shift_options):
     shifts
         A |NumPy array| containing a set of stable shift parameters.
     """
-    random_state = get_random_state(seed=shift_options['init_seed'])
+    rng = new_rng(0)
```

Comment:
I'm not sure I understand why we are not using `get_rng` here (and also a few other algorithms). If I understand correctly, if I get unlucky with a very bad random initialization I don't have any way to tell the algorithm to pick a new initialization for the same problem.

Reply:
I'm not using `get_rng` here, because in our discussions there seemed to be a desire (especially by @pmli) for determinism in the algorithm. Using `get_rng` would mean that subsequent calls to the algorithm with the same data would lead to different results (which I would be ok with). Of course, we could re-add the `seed` parameter to make the `0` configurable. However, I am fearing that would lead us down a rabbit hole where everywhere in pyMOR one might argue that a seed parameter is needed whenever something (implicitly) depends on randomness.

In general I would prefer the following policy: Either the algorithm crucially depends on randomness in the sense that different initial states lead to significantly different results. Then one should use `get_rng` and embrace this fact by allowing subsequent calls to lead to different results - with the possibility to reset the state from the outside if somehow desired. Or: the algorithm is essentially deterministic and its performance should not really depend on the random state. I.e. it will perform well almost surely, or something in that line. Then, fix some seed once and for all, like ARPACK does.

Reply:
I remember the ARPACK example, but I wasn't sure whether we made a definitive decision on this or not. The way you phrase the policy sounds good to me. Maybe at some point we can add a comment about this in the developer documentation or something similar.